drivers/wireless/lpwan/rn2483/rn2483.c

```diff
@@ -0,0 +1,1302 @@
+
+  if (filep->f_pos > 0)
+  {
+    return 0;
+  }
+
+  /* Get exclusive access */
+
+  err = nxmutex_lock(&priv->devlock);
+  if (err < 0)
+  {
+    return err;
+  }
+
+#ifndef CONFIG_DISABLE_PSEUDOFS_OPERATIONS
+  if (priv->unlinked)
+  {
+    /* Do not allow operations on unlinked drivers. */
+
+    nxmutex_unlock(&priv->devlock);
+    return 0;
+  }
+#endif
+
+  // TODO actually receive
+
+  // For now just get the radio version info
+  char command[] = "sys get ver\r\n";
+  length = file_write(&priv->uart, command, sizeof(command));
+  if (length < 0)
+  {
+    // TODO: log error
+    nxmutex_unlock(&priv->devlock);
+    return length;
+  }
+
+  // Read back the version string
+  length = file_read(&priv->uart, buffer, buflen);
+
+  filep->f_pos += length;
+  nxmutex_unlock(&priv->devlock);
+  return length;
+}
+
+/****************************************************************************
+ * Name: rn2483_write
+ *
+ * Description:
+ *     Write interface for transmitting over the RN2483.
+ ****************************************************************************/
+
+static ssize_t rn2483_write(FAR struct file *filep, FAR const char *buffer, size_t buflen)
+{ // // TODO actually transmit
+
+  FAR struct inode *inode = filep->f_inode;
+  FAR struct rn2483_dev_s *priv = inode->i_private;
+  int err;
+
+  err = nxmutex_lock(&priv->devlock);
+  if (err < 0)
+  {
+    return err;
+  }
+
+  return -ENOSYS;
+}
+
+/****************************************************************************
+ * Name: rn2483_ioctl
+ ****************************************************************************/
+
+static int rn2483_ioctl(FAR struct file *filep, int cmd, unsigned long arg)
+{
+  FAR struct inode *inode = filep->f_inode;
+  FAR struct rn2483_dev_s *priv = inode->i_private;
+  int err;
+
+  err = nxmutex_lock(&priv->devlock);
+  if (err < 0)
+  {
+    return err;
+  }
+
+#ifndef CONFIG_DISABLE_PSEUDOFS_OPERATIONS
+  if (priv->unlinked)
+  {
+    /* Do not allow operations on unlinked drivers. */
+
+    nxmutex_unlock(&priv->devlock);
+    return -ENODEV;
+  }
+#endif
+
+  /* Handle command */
+  switch (cmd)
+  {
+    
+    /*  Set Radio modulation */
+    case WLIOC_SETRADIOMOD:
+    {
+      enum rn2483_mod_e *ptr = (enum rn2483_mod_e *)(arg);
+      if (*ptr != RN2483_MOD_FSK && *ptr != RN2483_MOD_LORA)
+      {
+        return -EINVAL;
+      }
+      priv->config.mod = *ptr;
+
+      err = rn2483_radio_set_mod(priv, *ptr);
+      break;
+    }
+    /*  Get Radio modulation */
+    case WLIOC_GETRADIOMOD:
+    {
+      *((enum rn2483_mod_e *)(arg)) = priv->config.mod; // Returns modulation in pointer
+      break;
+    }
+
+
+    /*  Set Radio frequency */
+    case WLIOC_SETRADIOFREQ:
+    {
+      uint32_t *ptr = (uint32_t *)(arg);
+      if ((*ptr < 433000000 ||(*ptr > 434800000 && *ptr < 863000000) ||*ptr > 870000000))
+      {
+        return -EINVAL;
+      }
+      priv->config.freq = *ptr;
```

Comment:
One thing to keep in mind when setting your radio parameters is that you should store the updated config values _after_ you call the `radio_set*` function. You'll need to check the error response, because while you did check for a valid value, something could go horribly wrong with the radio (i.e. not responding). In that case, you've already updated the config which gives the user the impression that the configuration has taken effect before it has.

Example for frequency:
```c
// Your value checking code is above this
err = rn2483_radio_set_freq(priv, *ptr);
if (!err){
    priv->config.freq = *ptr; // This way we only update config on success
}
break;
```

Reply:
I just pushed code to add this.